akka-stream-tests/src/test/scala/akka/stream/scaladsl/FromMaterializationSpec.scala

```diff
@@ -226,24 +255,38 @@ class FromMaterializerSpec extends StreamSpec {
       val sink = Sink
         .fromMaterializer { (_, _) =>
           Sink.fromMaterializer { (_, attr) =>
-            Sink.fold(attr.nameLifted)(Keep.left)
+            Sink.cancelled.mapMaterializedValue(_ => attr.nameLifted)
```

Comment:
Using `Sink.fold` caused a timeout, not sure why.

Reply:
Curious, but not important here.

Reply:
do you mean that fold consistently fails, could it be caused by the changes in this PR?

Reply:
Well, the `fold`-based tests were working fine before, so it must be related to this PR. It sounds like this warrants a deeper look after all?

Reply:
I took a look at this, but didn't quite figure it out yet. For the new test `preserve attributes of inner sink` but with fold, the fold sink stage is created and started, sees a pull from downstream, but never sees upstream completing, so never emits and completes.

Looks like it fails the same way if I revert the change to the SetupSinkStage though, so potentially something tricky with the subOutlet/nested materialization that already was present.

Reply:
No, that was incorrect, I had changed the test Source a bit while investigating and forgotten about it. Only happens with the new materialization setup, I can see that the outer `SetupStage` `delegateToSubOutlet onUpstreamFinish`-call happens, then the materialization of the innermost sink, but the nested `SetupStage` never sees a `delegateToSubOutlet onUpstreamFinish`.

Reply:
Figured it out now, it the `SetupSinkStage` needs a `setKeepGoing(true)` or else it will complete once the inlet has completed (the materializer does not know about the substream).

Reply:
Thank you for looking into this! I made the change and reverted the tests.